frontend/src/app/chats/page.js

```diff
@@ -1,34 +1,176 @@
 "use client";
 
 import { useEffect, useState } from "react";
-import { useChatContext } from "@/context/ChatContextProvider";
-import { ChatWindow, ChatList } from "@/components";
+import { useChatContext, useChatDispatch } from "@/context/ChatContextProvider";
+import { ChatWindow, ChatList, ChatNotification } from "@/components";
 import { socket } from "@/lib";
 
 const Chats = () => {
+  const chatDispatch = useChatDispatch();
   const { clientPhoneNumber } = useChatContext();
+  const [chats, setChats] = useState([]);
+  const [aiMessages, setAiMessages] = useState([]);
+  const [newMessage, setNewMessage] = useState([]);
+  const [clients, setClients] = useState([]);
+  const [reloadChatList, setReloadChatList] = useState(false);
+  const [showWhisper, setShowWhisper] = useState([]);
+  const [loadingWhisper, setLoadingWhisper] = useState([]);
 
+  // Handle socketio
   useEffect(() => {
     socket.connect();
 
     function onConnect() {
-      console.log("FE Connected");
+      console.info("FE Connected");
     }
 
     function onDisconnect() {
-      console.log("FE Disconnected");
+      console.info("FE Disconnected");
+    }
+
+    function onChats(value) {
+      console.info(value, "socket chats");
+      if (value) {
+        const findClient = clients.find(
+          (c) =>
+            c.phone_number === value.conversation_envelope.client_phone_number
+        );
+        setReloadChatList(!findClient);
+
+        // to handle show & loading whisper
+        setShowWhisper((prev) => [
+          ...new Set([
+            ...prev,
+            value.conversation_envelope.client_phone_number,
+          ]),
+        ]);
+        setLoadingWhisper((prev) => [
```

Comment:
Can we simplify to something like this? `[{"conversationId": 1, "loading": true, "message": "something"}]` so setLoading can be: `{...prev.filter((x) => x.conversationId !== conversationId), {conversationId: new,  "message": null}`. What do you think?

Reply:
Yes, I think we can simplify that, instead of using multiple state.

Reply:
mantap